docs/reference/settings/security-settings.asciidoc

```diff
@@ -1947,6 +1947,311 @@ include::{es-repo-dir}/settings/common-defs.asciidoc[tag=ssl-supported-protocols
+Defaults to `100000`.
+
+// tag::jwt-http-connect-timeout-tag[]
+`http.connect_timeout` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Controls the behavior of the http client used for back-channel communication to
+the JWT Issuer endpoints. Specifies the timeout until a connection
+ is established. A value of zero means the timeout is not used. Defaults to `5s`.
+// end::jwt-http-connect-timeout-tag[]
+
+// tag::jwt-http-read-timeout-tag[]
+`http.connection_read_timeout` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Controls the behavior of the http client used for back-channel communication to
+the JWT Issuer endpoints. Specifies the timeout used when
+requesting a connection from the connection manager. Defaults to `5s`
+// end::jwt-http-read-timeout-tag[]
+
+// tag::jwt-http-socket-timeout[]
+`http.socket_timeout` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Controls the behavior of the http client used for back-channel communication to
+the JWT Issuer endpoints. Specifies the socket timeout (SO_TIMEOUT)
+in milliseconds, which is the timeout for waiting for data or, put differently,
+a maximum period inactivity between two consecutive data packets). Defaults to
+`5s`.
+// end::jwt-http-socket-timeout[]
+
+// tag::jwt-http-max-connections-tag[]
+`http.max_connections` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Controls the behavior of the http client used for back-channel communication to
+the JWT Issuer endpoints. Specifies the maximum number of
+connections allowed across all endpoints.
+// end::jwt-http-max-connections-tag[]
+
+// tag::jwt-http-max-endpoint-connections-tag[]
+`http.max_endpoint_connections` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Controls the behavior of the http client used for back-channel communication to
+the JWT Issuer endpoints. Specifies the maximum number of
+connections allowed per endpoint.
+// end::jwt-http-max-endpoint-connections-tag[]
+
+`http.proxy.scheme`::
+(<<static-cluster-setting,Static>>)
+Specifies the protocol to use to connect to the proxy server that will be
+used by the http client for all back-channel communication to the JWT Issuer
+endpoints. Defaults to `http`. Allowed values are `http` or `https`.
+
+`http.proxy.host`::
+(<<static-cluster-setting,Static>>)
+Specifies the address of the proxy server that will be used by the internal
+http client for all back-channel communication to the JWT Issuer
+endpoints. This includes requests to fetch the JSON Web Key Set from the
+JWT Issuer if `jwkset_path` is set as a URL.
+
+`http.proxy.port`::
+(<<static-cluster-setting,Static>>)
+Specifies the port of the proxy server that will be used by the http
+client for all backchannel communication to the JWT Provider
+endpoints. Defaults to `80`.
+
+[discrete]
+[[ref-jwt-ssl-settings]]
+===== JWT realm SSL settings
+// tag::jwt-ssl-description-tag[]
+The following settings can be used to configure SSL for all outgoing http connections
```

Comment:
Can we be more specific here about what the http connections are ( fetching jwkset )  ?